internal/graph/shadow_resolver.go

```diff
@@ -80,7 +82,6 @@ func (s ShadowResolver) ResolveCheck(ctx context.Context, req *ResolveCheckReque
 				}
 			}()
 
-			defer s.wg.Done()
 			ctx, cancel := context.WithTimeout(ctxClone, s.shadowTimeout)
 			defer cancel()
```

Comment:
Do we want the cancel above the recover as well?

Reply:
No, we want the cancel to be inside the recover (in case the cancel causing panic)